db/db_impl.cc

```diff
@@ -599,7 +613,64 @@ void DBImpl::StartTimedTasks() {
             stats_dump_period_sec * 1000000));
       }
     }
+    stats_persist_period_sec = mutable_db_options_.stats_persist_period_sec;
+    if (stats_persist_period_sec > 0) {
+      if (!thread_persist_stats_) {
+        thread_persist_stats_.reset(new rocksdb::RepeatableThread(
+          [this]() { DBImpl::PersistStats(); }, "pst_st", env_,
+          stats_persist_period_sec * 1000000));
+      }
+    }
+  }
+}
+
+void DBImpl::PersistStats() {
+  TEST_SYNC_POINT("DBImpl::PersistStats:1");
+#ifndef ROCKSDB_LITE
+  const DBPropertyInfo* cf_property_info =
+      GetPropertyInfo(DB::Properties::kCFStats);
+  assert(cf_property_info != nullptr);
+  const DBPropertyInfo* db_property_info =
+      GetPropertyInfo(DB::Properties::kDBStats);
+  assert(db_property_info != nullptr);
+  std::map<std::string, std::string> stats_map;
+  if (shutdown_initiated_) {
+    return;
+  }
+  {
+    InstrumentedMutexLock l(&mutex_);
+    default_cf_internal_stats_->GetMapProperty(
+        *db_property_info, DB::Properties::kDBStats, &stats_map);
+    for (auto cfd : *versions_->GetColumnFamilySet()) {
+      if (cfd->initialized()) {
+        cfd->internal_stats()->GetMapProperty(
+            *cf_property_info, DB::Properties::kCFStatsNoFileHistogram,
+            &stats_map);
+      }
+    }
+    for (auto cfd : *versions_->GetColumnFamilySet()) {
+      if (cfd->initialized()) {
+        cfd->internal_stats()->GetMapProperty(
+            *cf_property_info, DB::Properties::kCFFileHistogram, &stats_map);
+      }
+    }
+  }
+  TEST_SYNC_POINT("DBImpl::PersistStats:2");
+  ROCKS_LOG_WARN(immutable_db_options_.info_log,
+                 "------- PERSISTING STATS -------");
+  WriteOptions wo;
+  ColumnFamilyOptions cf_options;
+  const uint64_t now_micros = env_->NowMicros();
+  std::string now_micros_string = std::to_string(now_micros);
+  int keycount = 0;
+  for (auto iter = stats_map.begin(); iter != stats_map.end(); ++iter) {
+    // TODO(Zhongyi) use more readable timestamp?
```

Comment:
I find this: https://github.com/facebook/rocksdb/blob/master/env/posix_logger.h#L111